src/InsolationCalc.jl

```diff
@@ -1,60 +1,264 @@
+The insolation is averaged over a full day.
+
+# Arguments
+- `date::DateTime`: Current date
+- `latitude::FT`: Latitude [degrees]
+- `param_set::IP.AIP`: Parameter struct
+- `orbital_data::Union{OrbitalDataSplines, Nothing}`: (default nothing) Orbital parameter splines.
+  **Required** when `milankovitch=true` for GPU compatibility.
+- `milankovitch::Bool`: (default false) Use time-varying orbital parameters (Milankovitch cycles)
+- `solar_variability::Bool`: (default false) Use time-varying solar luminosity (placeholder for future)
+
+# Returns
+- `F`: Daily averaged TOA insolation [W m⁻²]
+- `S`: Solar flux [W m⁻²]
+- `μ`: Daily averaged cosine of solar zenith angle [unitless]
+
+# Examples
+```julia
+# Modern climate (fixed epoch parameters)
+F, S, μ = daily_insolation(date, lat, param_set)
+
+# Paleoclimate with Milankovitch cycles (GPU-compatible)
+od = OrbitalDataSplines()  # Load once on CPU
+F, S, μ = daily_insolation(date, lat, param_set; orbital_data=od, milankovitch=true)
+```
+
+# GPU Usage
+For GPU execution, pre-load orbital data on CPU:
+```julia
+od = OrbitalDataSplines()  # CPU only, do this once
+# In GPU kernel:
+F, S, μ = daily_insolation(date, lat, param_set; orbital_data=od, milankovitch=true)
+```
+"""
+function daily_insolation(
+    date::DateTime,
+    latitude::FT,
+    param_set::IP.AIP;
+    orbital_data::Union{OrbitalDataSplines, Nothing} = nothing,
+    milankovitch::Bool = false,
+    solar_variability::Bool = false,
+) where {FT <: Real}
+    # Get orbital parameters - GPU-friendly approach
+    # Separate data loading (expensive, done once) from parameter selection
+
+    # Compute epoch parameters (always needed as fallback)
+    ϖ_epoch, γ_epoch, e_epoch = Insolation.orbital_params(param_set)
+
+    # Compute time-varying parameters if needed
+    if milankovitch
+        # Require pre-loaded orbital data for GPU compatibility
+        if isnothing(orbital_data)
+            error(
+                "orbital_data must be provided when milankovitch=true for GPU compatibility. " *
+                "Load OrbitalDataSplines on CPU first: od = OrbitalDataSplines(); " *
+                "then call daily_insolation(...; orbital_data=od, milankovitch=true)",
+            )
+        end
+        Δt_years = Insolation.years_since_epoch(param_set, date)
+        ϖ_milan, γ_milan, e_milan =
+            Insolation.orbital_params(orbital_data, Δt_years)
+    else
+        # Use dummy values (won't be selected)
+        ϖ_milan, γ_milan, e_milan = ϖ_epoch, γ_epoch, e_epoch
     end
-    μ = cos(θ)
-    # TOA solar flux (3.12)
-    S = S0 * (d0 / d)^2
-    # return inputs needed for RRTMGP.jl
-    return S, μ
+
+    # GPU-friendly parameter selection using ifelse (no divergent branching)
+    ϖ = ifelse(milankovitch, FT(ϖ_milan), FT(ϖ_epoch))
+    γ = ifelse(milankovitch, FT(γ_milan), FT(γ_epoch))
+    e = ifelse(milankovitch, FT(e_milan), FT(e_epoch))
+    orb_params = (ϖ, γ, e)
```

Comment:
Isn't this the same as `orb_params = (FT(ϖ_milan), FT(γ_milan), FT(e_milan))` since you have an ifelse before?